pkg/multi-tenancy/write/write_test.go

```diff
@@ -0,0 +1,66 @@
+package write
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/timescale/promscale/pkg/multi-tenancy/config"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+	"github.com/timescale/promscale/pkg/prompb"
+)
+
+func TestMultiTenancyWrite(t *testing.T) {
+	var (
+		lblsArr    = getlbls()
+		tenantName = "tenant-a" // Assume from TENANT header.
+	)
+
+	// With valid tenants.
+	conf := config.NewSelectiveTenancyConfig([]string{"tenant-a", "tenant-b"}, false)
+	authr := NewAuthorizer(conf)
+	require.True(t, authr.IsAuthorized(tenantName))
+	for _, lbls := range lblsArr {
+		lb, ok := authr.VerifyAndApplyTenantLabel(tenantName, lbls)
+		require.True(t, ok)
+		require.True(t, containsAppliedTenantLabel(lb))
+	}
+
+	// Should not verify.
+	conf = config.NewSelectiveTenancyConfig([]string{"tenant-b"}, false)
+	authr = NewAuthorizer(conf)
+	require.False(t, authr.IsAuthorized(tenantName))
+
+	// Empty tenant write.
+	conf = config.NewOpenTenancyConfig(false)
+	authr = NewAuthorizer(conf)
+	require.False(t, authr.IsAuthorized(""))
+
+	conf = config.NewOpenTenancyConfig(true)
+	authr = NewAuthorizer(conf)
+	require.True(t, authr.IsAuthorized(""))
+}
+
+func containsAppliedTenantLabel(lbls []prompb.Label) bool {
+	for _, lbl := range lbls {
+		if lbl.Name == config.TenantLabelKey {
+			return true
+		}
+	}
+	return false
+}
+
+func getlbls() [][]prompb.Label {
```

Comment:
should add case with tenant label already set. Also should test case where it's set to the wrong tenant

Reply:
done in recent commit